app/src/main/java/ch/epfl/cs311/wanderwave/model/repository/FirebaseRepository.kt

```diff
@@ -21,7 +22,7 @@ interface FirebaseRepository<T> {
 
   fun deleteItem(itemId: String)
 
-  fun getItem(item: T): Flow<T>
-
   fun getItem(itemId: String): Flow<T>
+
+  fun getItem(itemId: String, onSuccess: (DocumentSnapshot, MutableStateFlow<T?>) -> Unit): Flow<T>
```

Comment:
As written before somewhere, why the onSuccess?

Reply:
This is because of the refactoring, in the firebase repository, I have only the base function and I have the beacon and profile connection that sometimes need to do something after the success for example fetch all the references for tracks.

Reply:
But still, when you need to be able to do something on success, you should just be able to await the result, do whatever the onSuccess would do with the result, and pass it on to another flow if necessary, no?

Reply:
No because you don't have all the information in the document snapshot and namely the references to other documents are not stored in the flow of any of the data classes.